BlogTemplate/Pages/Edit.cshtml.cs

```diff
@@ -92,6 +96,8 @@ public class EditedPostModel
             [Required]
             public string Body { get; set; }
             public string Excerpt { get; set; }
+            [Required]
+            public int Id { get; set; }

```

Comment:
You don't want this on the ViewModel.  You're already getting it from the route when you need it.  Instead, you'll want to pass this to the view in a different way (maybe in TempData or ViewData).